src/engine/loopingcontrol.h

```diff
@@ -32,14 +32,14 @@ class LoopingControl : public EngineControl {
     // process() updates the internal state of the LoopingControl to reflect the
     // correct current sample. If a loop should be taken LoopingControl returns
     // the sample that should be seeked to. Otherwise it returns currentSample.
-    double process(const double dRate,
+    virtual double process(const double dRate,
```

Comment:
yes that's kind of ugly but it's a problem endemic to C++ -- we can't just duck-type everything away.  Since tests are so critical, it's worth having a bit of ugliness to keep it well-tested.